### DISCUSSION TITLE
Realtime API voice chat

### DISCUSSION THREAD

**Discussion by matari** [cline/cline#479] [Feature Requests]
Realtime API voice chat with file or folder context.

**Comment by dbsxdbsx** [comment]
I really want this feature too!

But the most important thing is that the AI should truly understand the current workspace context, whether it’s from the README, the actual code files(especially those custom-named structs/methods/fucntions), or related SDKs. This way, when users speak, the AI can understand specific terminology and give more accurate help. 

Also, if the AI doesn’t fully understand what the user says in voice mode, it should ask follow-up questions instead of guessing.

**Comment by bdcdo** [comment]
https://github.com/ILikeAI/AlwaysReddy is written in python, but might still be a good start for implementing this feature. At least for me, just being able to click a button and give further instructions to Cline would be a great quality of life improvement
